services/venice-server/src/main/java/com/linkedin/venice/listener/StorageReadRequestsHandler.java

```diff
@@ -397,8 +398,12 @@ private ReadResponse handleSingleGetRequest(GetRouterRequest request) {
     response.setValueRecord(valueRecord);
 
     if (keyValueProfilingEnabled) {
-      response.setOptionalKeySizeList(Optional.of(Arrays.asList(key.length)));
-      response.setOptionalValueSizeList(Optional.of(Arrays.asList(response.isFound() ? valueRecord.getDataSize() : 0)));
+      IntList keyList = new IntArrayList(1);
+      keyList.set(0, key.length);
+      response.setKeyListSize(keyList);
+      IntList valueList = new IntArrayList(1);
+      valueList.set(0, response.isFound() ? valueRecord.getDataSize() : 0);
+      response.setValueListSize(valueList);
```

Comment:
I think we could use `IntLists::singleton` here.

Reply:
Sure, did not know it has singleton method.

Reply:
Cool, let's use it 👌